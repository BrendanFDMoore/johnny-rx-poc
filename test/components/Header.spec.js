import React from 'react';
import { mount } from 'enzyme';

const mockDispatch = jest.fn();
jest.mock('react-router/Link', () =>
  (({ to, children }) => (<a href={to}>{children}</a>))
);

const Header = require('components/Header').default;

function setup() {
  const props = {
    dispatch: mockDispatch,
    user: {
      isAuthenticated: false
    }
  };

  return mount(<Header {...props} />);
}

describe('Header', () => {
  const wrapper = setup();

  it('should be a StatelessComponent', () => {
    expect(wrapper.instance().constructor.name).toBe('StatelessComponent');
  });

  it('should render properly', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should handle login click', () => {
    wrapper.find('.app__header__login').simulate('click');
    expect(mockDispatch.mock.calls[0][0]).toEqual({ type: 'USER_LOGIN_REQUEST' });
  });

  it('should handle logout click', () => {
    wrapper.setProps({
      user: {
        isAuthenticated: true
      }
    });

    wrapper.find('.app__header__logout').simulate('click');
    expect(mockDispatch.mock.calls[1][0]).toEqual({ type: 'USER_LOGOUT_REQUEST' });
  });
});

