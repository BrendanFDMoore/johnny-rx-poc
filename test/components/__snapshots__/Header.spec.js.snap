exports[`Header should render properly 1`] = `"<header class=\"app__header\"><div class=\"app__container\"><a href=\"/\" class=\"app__header__logo\"><div class=\"app__logo\"><span class=\"isvg failed\"><img src=\"\" alt=\"React-Redux-Observables Boilerplate\"></span></div></a><h1>React-Redux-Observables Boilerplate</h1><div class=\"app__header__menu\"><ul class=\"list-unstyled\"><li><a href=\"#logout\" class=\"app__header__logout\"><i class=\"i-sign-out\"></i><span>logout</span></a></li></ul></div></div></header>"`;
